basiclibrary/src/test/java/basiclibrary/LibraryTest.java

```diff
@@ -20,44 +22,79 @@ public static boolean checkRollRange() {
-        assertFalse("{1, 2, 3} should not return true", classUnderTest.containsDuplicates(new int[] {1, 2, 3}));
+        assertTrue("{1, 2, 3, 2, 5} should return 'true'", Library.containsDuplicates(testArray));
+        testArray = new int[] {1, 2, 3};
+        assertFalse("{1, 2, 3} should not return true", Library.containsDuplicates(testArray));
     }
+
     @Test public void testCalculatesAverages() {
-        Library classUnderTest = new Library();
         int[] testArray = {1, 23, 55, 4, 5};
         float averageOfTestArray = (float) ((1 + 23 + 55 + 4 + 5) / 5.0);
-        //Happy path:
-        assertTrue("{1, 23, 55, 4, 5} should return float ~17.6", (Math.abs(classUnderTest.calculatesAverages(testArray) - averageOfTestArray) < 0.1));
-        //Expected failure:
-        assertFalse("calculatesAverages() should not return an int", (classUnderTest.calculatesAverages(testArray) == (int) classUnderTest.calculatesAverages(testArray)));
+        boolean actual = (Math.abs(Library.calculatesAverages(testArray) - averageOfTestArray) < 0.1);
+        boolean actual2 = (Library.calculatesAverages(testArray) == (int) Library.calculatesAverages(testArray));
+
+        //Happy paths:
+        assertTrue("{1, 23, 55, 4, 5} should return float ~17.6", actual);
+        assertFalse("calculatesAverages() should not return an int", actual2);
     }
     @Test public void testArrayWithLowestAverage() {
-        Library classUnderTest = new Library();
         int[][] weeklyMonthTemperatures = {
                 {66, 64, 58, 65, 71, 57, 60},
                 {57, 65, 65, 70, 72, 65, 51},
                 {55, 54, 60, 53, 59, 57, 61},
                 {65, 56, 55, 52, 55, 62, 57}
         };
-        //Happy path:
-        assertTrue("the third array in weeklyMonthTemperatures has the lowest average", (classUnderTest.arrayWithLowestAvg(weeklyMonthTemperatures) == weeklyMonthTemperatures[2]));
-        //Expected failure:
-        assertFalse("arrayWithLowestAvg() should not return the average number, just an array", (!classUnderTest.arrayWithLowestAvg(weeklyMonthTemperatures).getClass().isArray()));
+        boolean actual = (Library.arrayWithLowestAvg(weeklyMonthTemperatures) == weeklyMonthTemperatures[2]);
+        boolean actual2 = (!Library.arrayWithLowestAvg(weeklyMonthTemperatures).getClass().isArray());
+
+        //Happy paths:
+        assertTrue("the third array in weeklyMonthTemperatures has the lowest average", actual);
+        assertFalse("arrayWithLowestAvg() should not return the average number, just an array", actual2);
     }
 
+    @Test public void testMapTemperatures() {
+        int[][] weeklyMonthTemperatures = {
+                {66, 64, 58, 65, 71, 57, 60},
+                {57, 65, 65, 70, 72, 65, 51},
+                {55, 54, 60, 53, 59, 57, 61},
+                {65, 56, 55, 52, 55, 62, 57}
+        };
+        //Happy Paths:
+        assertTrue(Library.mapTemperatures(weeklyMonthTemperatures).contains("63"));
```

Comment:
A good rule of thumb is to have one assert per test, so if one fails, you know why. However, you can also test one concept per test... which is more what you are doing. Either way, decide consciously how many asserts you can have without causing confusion when failing tests return.